crates/duty-tracker/src/contract_manager.rs

```diff
@@ -398,51 +382,30 @@ impl ContractManager {
                             // this could be a transient issue, so no need to break immediately
                         }
                     },
+                    instant = interval.tick() => {
+                        debug!(?instant, "constructing nags");
 
-                    // Finally we pull from the duties channel and execute those.
-                    Some((contract_txid, duty_result)) = duty_response_receiver.recv() => {
-                        match duty_result {
-                            Ok(contract_duties) if !contract_duties.is_empty() => {
-                                info!(%contract_txid, num_duties=%contract_duties.len(), "received duties from contract actor");
-                                duties.extend(contract_duties);
-                            },
-                            Ok(_) => {
-                                trace!(%contract_txid, "contract actor processed event with no duties");
-                            },
-                            Err(e) => {
-                                error!(%contract_txid, %e, "contract actor failed to process event");
-                                // For aggregate signatures, this is catastrophic
-                                // Check if this was an aggregate signature event and break if so
-                                // For now, we'll log the error and continue
-                            }
-                        }
-                    },
-
-                    _instant = interval.tick() => {
-                        info!("nagging peers for necessary p2p messages");
-                        let nags = ctx.nag().await;
+                        let nags = ctx.nag();
                         for nag in nags {
                             p2p_handle.send_command(nag).await;
                         }
-
-                        // Periodically cleanup completed contracts
-                        ctx.state.active_contracts.cleanup_completed_contracts().await;
                     }
                 }
 
-                for duty in duties {
-                    info!(%duty, "starting duty execution from new events");
-                    if execute_duty(ctx.cfg.clone(), output_handles.clone(), duty.clone())
-                        .await
-                        .is_err()
-                    {
-                        break;
-                    }
-                }
+                duties.into_iter().for_each(|duty| {
+                    debug!(%duty, "starting duty execution from new events");
+
+                    let cfg = ctx.cfg.clone();
+                    let output_handles = output_handles.clone();
+                    task::spawn(async move {
+                        if let Err(e) = execute_duty(cfg, output_handles, duty.clone()).await {
+                            error!(%e, %duty, "failed to execute duty");
+                        }
+                    });
```

Comment:
Yep also don't do a task spawn here either.

Reply:
Done in 8e258e4876241d947fab9769b2844657e9be92cc